src/components/Canvas/Editor.tsx

```diff
@@ -1388,54 +1388,6 @@ export default class Editor extends EventDispatcher {
     }
   }
 
-  fillMissingIndices(previousCoord: Coord, currentCoord: Coord) {
-    if (!previousCoord || !currentCoord) return;
-
-    const gridIndices = this.dataLayer.getGridIndices();
-    const pixelIndex = getPixelIndexFromMouseCartCoord(
-      currentCoord,
-      this.getRowCount(),
-      this.getColumnCount(),
-      this.gridSquareLength,
-      gridIndices.topRowIndex,
-      gridIndices.leftColumnIndex,
-    );
-
-    if (
-      Math.abs(currentCoord.x - previousCoord.x) >= this.gridSquareLength ||
-      Math.abs(currentCoord.y - previousCoord.y) >= this.gridSquareLength
-    ) {
-      const previousIndex = getPixelIndexFromMouseCartCoord(
-        previousCoord,
-        this.getRowCount(),
-        this.getColumnCount(),
-        this.gridSquareLength,
-        gridIndices.topRowIndex,
-        gridIndices.leftColumnIndex,
-      );
-      if (!previousIndex) return;
-
-      if (
-        (previousIndex &&
```

Comment:
I think the previous issue of the interaction layer not coloring pixels using bressenham's algorithm when the mouse goes out of the pixel grid was due to this part. Previously, I remember that you used `fillMissingIndices(this.mouseMoveWorldPos, this.previousMouseMovePos)`. Since `this.mouseMoveWorldPos` was wrongly used as a previousCoord (it should have been the currentCoord!), the `if` condition here must have been passed.

Reply:
Appreciate this detailed analysis. Yes this was a mistake